From 5e20f92cdeda097609118362264bc1a29225a02e Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Thu, 13 Feb 2014 14:29:22 +0800
Subject: [PATCH 2/2] ENGR00214328 fix CTS
 android.provider.cts.MediaStore_FilesTest#testAccess
 failed.

Revert "ENGR00280148 CTS android.provider.cts.MediaStore_Images_MediaTest failed"

This reverts commit 98068e63339f9f18c817b840380c75a1ef327885.

This will cause android.provider.cts.MediaStore_FilesTest#testAccess failure.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 src/com/android/providers/media/MediaProvider.java | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/com/android/providers/media/MediaProvider.java b/src/com/android/providers/media/MediaProvider.java
index 9e50392..4528c56 100755
--- a/src/com/android/providers/media/MediaProvider.java
+++ b/src/com/android/providers/media/MediaProvider.java
@@ -2766,9 +2766,7 @@ public class MediaProvider extends ContentProvider {
             values = initialValues;
         }
 
-        if (!ensureFileExists(file)) {
-            throw new IllegalStateException("Unable to create new file: " + file);
-        }
+        // we used to create the file here, but now defer this until openFile() is called
         return values;
     }
 
-- 
1.8.0

